### PR TITLE
Initial commit for build journal. We will use tools/saftw.py to gener…

### PR DESCRIPTION
**Pull request by zmallen**
…ate the journal, which instantiates a testrunner object and calls the build journal function. We initiate a sqlite connection to write results out to the sqlite database

### CODE REVIEW

**Comment by zmallen** [comment]
Right now a user runs `tools/build_journal.py` with different arguments to build a journal file with a table entry to log FTW tests. It can be optimized further because it connects to the db for every test instead of every run, going to spawn ftw/modsec locally for testing

File: tools/saftw.py [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+                    print e
+                    pass
+                if co.debug == 1:
+                    print "Executed test {}".format(test.test_title)
+                jrnl.add_entry(now, test.test_title, logtag, invert, hua.response_object)
+    return 0
+    
+if __name__ == "__main__":
+    main()
+0
```
Comment (csjperon) [inline tools/saftw.py]: Not sure what this 0 is doing here (or if was part of the original code) but we could probably get rid of it

File: tools/saftw.py [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+
+    def match(self, rule, ln):
+        m = self.prog.match(ln)
+        if m == None:
+            return False
+        if int(m.group(1)) == rule:
+            return True
+        return False
+
+class varnish_waf(waf_plugin):
```
Comment (csjperon) [inline tools/saftw.py]: We should probably get rid of this, since the logging format would be user defined anyway

**Comment by zmallen** [comment]
I dont think I can merge until I write some tests, which ill get to tonight/tomorrow

**Comment by zmallen** [comment]
It also needs a readme update for writing an integration /using the tool

**Comment by csjperon** [comment]
Alright, I might get a chance to test before now and then

**Comment by zmallen** [comment]
@csanders-git

**Pull request merged**